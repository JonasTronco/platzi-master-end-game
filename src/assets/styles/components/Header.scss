@import '../main.scss';

.header {
  @include flexCenCen;
  width: 100%;
  height: 75px;
  background-color: $primary_color;
  color: $white_color;
  font-family: $title_font;

  &__logo {
    width: 45px;
    height: 45px;
    margin-right: $space_x3;
    fill: $white_color;
  }
}
