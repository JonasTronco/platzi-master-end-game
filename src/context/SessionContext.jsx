import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
