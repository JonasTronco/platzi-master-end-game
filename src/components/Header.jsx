import '../assets/styles/components/Header.scss';
import Logo from '../assets/icons/logo.svg';

function Header() {
  return (
    <header className="header">
      <Logo className="header__logo" />
      <strong className="header__text">
        HALAH
        <br />
        Laboratories
      </strong>
    </header>
  );
}

export default Header;
