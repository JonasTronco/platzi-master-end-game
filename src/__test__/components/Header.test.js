import Header from '../../components/Header';
import { create } from 'react-test-renderer';
import { SessionProvider } from '../../context/SessionContext';
describe('Header Component', () => {
  test('Should render Header component', () => {
    const component = create(
      <SessionProvider>
        <Header />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
