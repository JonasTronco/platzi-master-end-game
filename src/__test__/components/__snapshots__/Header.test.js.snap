// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should render Header component 1`] = `
<header
  className="header"
>
  <div
    className="header__logo"
  />
  <strong
    className="header__text"
  >
    HALAH
    <br />
    Laboratories
  </strong>
</header>
`;
